src/snowflake/snowpark/async_job.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+from enum import Enum
+from typing import List
+
+import snowflake.snowpark
+from snowflake.connector import SnowflakeConnection
+from snowflake.connector.cursor import ResultMetadata
+from snowflake.snowpark._internal.utils import result_set_to_iter, result_set_to_rows
+
+
+class AsyncDataType(Enum):
+    PANDAS = "pandas"
+    ROW = "row"
+    ITERATOR = "iterator"
+    PANDAS_BATCH = "pandas_batch"
```

Comment:
Do users need to use this Enum? If no we better hide it.

Reply:
Do you mean that I should put this in other place like _internal.utils ?